File: Kata.FizzBuzz/FizzBuzz.cs
using System.Collections.Generic;
using System.Globalization;
using System.Linq;

namespace Kata.FizzBuzz
{
    public class FizzBuzz
    {
        public const string FIZZBUZZ = "FIZZBUZZ";
        public const string FIZZ = "FIZZ";
        public const string BUZZ = "BUZZ";

        public string Parse(int input)
        {
            if (input % 3 == 0 && input % 5 == 0)
            {
                return FIZZBUZZ;
            }
            else if (input%3==0)
            {
                return FIZZ;
            }
            else if (input%5 == 0)
            {
                return BUZZ;
            }
            return input.ToString(CultureInfo.InvariantCulture);
        }

        public IEnumerable<string> Generate()
        {
            return Enumerable.Range(1, 100).Select(Parse);
        }
    }
}

File: Kata.FizzBuzz.Swift/Kata/FizzBuzz.swift
class FizzBuzz {
    func parse(number:Int) -> String {
        
        if number <= 0 {
            return "\(number)"
        }
        
        switch (number % 3, number % 5) {
        case (0, 0):
            return "FizzBuzz"
        case (0, _):
            return "Fizz"
        case (_, 0):
            return "Buzz"
        default:
            return "\(number)"
        }
        
        
    }
    
    func parseAllNumbers(numbers: [Int]) -> [String] {
        return numbers.map(parse)
    }
}